pkg/virt-launcher/virtwrap/sriov/hostdev_test.go

```diff
@@ -0,0 +1,252 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package sriov_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/sriov"
+
+	v1 "kubevirt.io/client-go/api/v1"
+)
+
+var _ = Describe("SRIOV HostDevice", func() {
+	It("creates no device given no interfaces", func() {
+		vmi := &v1.VirtualMachineInstance{}
+
+		Expect(sriov.CreateHostDevices(vmi)).To(BeEmpty())
+	})
+
+	It("creates no device given no SRIOV interfaces", func() {
+		iface := v1.Interface{}
+		iface.Masquerade = &v1.InterfaceMasquerade{}
+		vmi := &v1.VirtualMachineInstance{}
+		vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{iface}
+
+		Expect(sriov.CreateHostDevices(vmi)).To(BeEmpty())
+	})
+
+	It("fails to create device given no available host PCI", func() {
+		iface := newSRIOVInterface("test")
+		vmi := &v1.VirtualMachineInstance{}
+		vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{iface}
+		_, err := sriov.CreateHostDevices(vmi)
+
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("creates no device given bad host PCI address", func() {
```

Comment:
IIUC, shouldn't this be phrased with fail like so: `fails to create device given bad host PCI address`?